src/apps/main/antivirus/ManualSystemScan.ts

```diff
@@ -0,0 +1,229 @@
+    }
+    this.totalScannedFiles++;
+
+    const progressValue = this.totalItemsToScan > 0 ? Math.round((this.totalScannedFiles / this.totalItemsToScan) * 100) : 0;
+
+    const progressEvent: ProgressData = {
+      currentScanPath: file,
+      infectedFiles: this.infectedFiles,
+      progress: progressValue,
+      totalScannedFiles: this.totalScannedFiles,
+    };
+    this.progressEvents.push(progressEvent);
+  };
+
+  public stopScan = async () => {
+    this.cancelled = true;
+    this.scanSessionId++;
+    if (this.manualQueue) {
+      console.log('KILLING PROCESSES');
+      this.manualQueue.kill();
+    }
+
+    if (this.antivirus) {
+      await this.antivirus.stopClamAv();
+    }
+
+    this.resetCounters();
+  };
+
+  private finishScan(currentSession: number) {
+    this.cancelled = true;
+    if (currentSession !== this.scanSessionId) return;
+
+    if (this.progressEvents.length > 0) {
+      eventBus.emit('ANTIVIRUS_SCAN_PROGRESS', {
+        ...(this.progressEvents.pop() as ProgressData),
+        done: true,
+      });
+    }
+    this.resetCounters();
+  }
+
+  private resetCounters() {
+    this.totalInfectedFiles = 0;
+    this.totalScannedFiles = 0;
+    this.infectedFiles = [];
+    this.progressEvents = [];
+    this.totalItemsToScan = 0;
+    this.manualQueue = null;
+    this.antivirus = null;
+  }
+
+  private handlePreviousScannedItem = async (currentSession: number, scannedItem: ScannedItem, previousScannedItem: ScannedItem) => {
+    if (currentSession !== this.scanSessionId) return;
+
+    if (scannedItem.updatedAtW === previousScannedItem.updatedAtW || scannedItem.hash === previousScannedItem.hash) {
+      this.trackProgress(currentSession, {
+        file: previousScannedItem.pathName,
+        isInfected: previousScannedItem.isInfected,
+      });
+      return;
+    }
+  };
+
+  public async scanItems(pathNames?: string[]): Promise<void> {
+    this.cancelled = false;
+    this.scanSessionId++;
+    const currentSession = this.scanSessionId;
+
+    if (!this.antivirus) {
+      this.antivirus = await Antivirus.createInstance();
+    }
+    const antivirus = this.antivirus;
+
+    let reportProgressInterval: NodeJS.Timeout | null = null;
+
+    reportProgressInterval = setInterval(() => {
+      if (this.progressEvents.length > 0) {
+        eventBus.emit('ANTIVIRUS_SCAN_PROGRESS', {
+          ...(this.progressEvents.pop() as ProgressData),
+        });
+        this.progressEvents = [];
+      }
+    }, 1000);
+
+    console.time('manual-scan');
+
+    const scan = async (filePath: string) => {
+      if (this.cancelled) return;
+      console.log('SCAN ITEM: ', filePath);
+      try {
+        const scannedItem = await transformItem(filePath);
+        const previousScannedItem = await this.dbConnection.getItemFromDatabase(scannedItem.pathName);
+        if (previousScannedItem) {
+          await this.handlePreviousScannedItem(currentSession, scannedItem, previousScannedItem);
```

Comment:
Return directly here, the `await` + `return` is not needed. If you want to catch the error inside, `return await` then